src/test/compile-fail/issue-23046.rs

```diff
@@ -10,15 +10,15 @@
 
 pub enum Expr<'var, VAR> {
     Let(Box<Expr<'var, VAR>>,
-        Box<for<'v: 'var> Fn(Expr<'v, VAR>) -> Expr<'v, VAR> + 'var>)
```

Comment:
Looks like the use of bounds was intentional here (https://github.com/rust-lang/rust/issues/23046).

Reply:
Need to check whether they are totally ignored during lowering to `ty` representation or just not enforced properly (and still may cause ICEs or something).

Reply:
Okay I will bring the bound back. So far I haven't observed anything that would react to these bounds, and AFAIK @eddyb also said they are ignored.

Are you saying I should check the lowering code whether they are ignored? I'm not familiar with that code at all, not even sure where it lives, but I can certainly try.^^